src/ai/backend/agent/docker/agent.py

```diff
@@ -1487,7 +1487,11 @@ async def push_image(self, image_ref: ImageRef, registry_conf: ImageRegistry) ->
             }
 
         async with closing_async(Docker()) as docker:
-            await docker.images.push(image_ref.canonical, auth=auth_config)
+            # TODO: Remove this useless type casting after resolving https://github.com/aio-libs/aiodocker/pull/909.
+            result = cast(list, await docker.images.push(image_ref.canonical, auth=auth_config))
+
+            if error := result[-1].get("error"):
+                raise RuntimeError(f"Failed to push image: {error}")
```

Comment:
`docker query` won't return an empty list, right?

Reply:
I think instead of making assumptions about how this API work, it would be better to add exception handling.
Thank you for the review!